pkg/model/project.go

```diff
@@ -243,3 +350,73 @@ func (p *ProjectSSOConfig_GitHub) GenerateAuthCodeURL(project, callbackURL, stat
 
 	return authURL, nil
 }
+
+// UpdateRBACRoles update only custom rbac roles.
+func (p *Project) UpdateRBACRoles(roles []*ProjectRBACRole) error {
+	v := make(map[string]struct{}, len(roles))
+	update := make([]*ProjectRBACRole, 0, len(roles))
+	for _, role := range roles {
+		if role.IsBuiltin {
+			continue
+		}
+		if role.IsBuiltinName() {
+			return fmt.Errorf("cannot use built-in role name")
+		}
+		if _, ok := v[role.Name]; ok {
+			return fmt.Errorf("role name must be unique")
+		}
+		v[role.Name] = struct{}{}
+		update = append(update, role)
+	}
```

Comment:
How about moving this to a new `ValidateRBACRoles` function?
After that, we can assign the validated data to `p.RbacRoles` directly instead of using this `UpdateRBACRoles` function.

Reply:
That's sound good.
I addressed https://github.com/pipe-cd/pipecd/pull/3477/commits/5ee990ef9de80d64541e6ab749ebdcb88b9ab606.